internal/pkg/cli/env_init.go

```diff
@@ -257,10 +265,23 @@ func (o *initEnvOpts) Ask() error {
 
 // Execute deploys a new environment with CloudFormation and adds it to SSM.
 func (o *initEnvOpts) Execute() error {
-	o.initRuntimeClients()
+	if err := o.initRuntimeClients(); err != nil {
+		return err
+	}
+	appVersion, err := o.appVersionGetter.Version()
+	if err != nil {
+		return fmt.Errorf("get template version of application %s: %w", o.appName, err)
+	}
+	if diff := semver.Compare(appVersion, o.templateVersion); diff > 0 && !o.allowAppDowngrade {
```

Comment:
i wonder if `o.allowAppDowngrade` should skip this logic entirely? in case `Version()` is returning an error due to a messed up template, it would allow you to get around that issue